### PR TITLE
continued warehouse loader refactor

### PR DESCRIPTION
**Pull request by snopoke**
##### SUMMARY
Building on https://github.com/dimagi/commcare-hq/pull/25412

Core changes here are:
* distinguish between loading dependencies and sql template context vars

### CODE REVIEW

**Review by mkangia — APPROVED** [review]

File: corehq/apps/app_manager/tests/util.py [hunk 1]
```diff
@@ -173,15 +174,13 @@ def commtrack_enabled(is_enabled):
...
+    for doc_type in app_doc_types():
+        res = Application.get_db().view(
+            'all_docs/by_doc_type',
+            startkey=[doc_type],
+            endkey=[doc_type, {}],
+            reduce=False,
+            include_docs=True
+        )
+        for row in res:
+            Application.get_db().delete_doc(row['doc'])
```
Comment (mkangia) [inline corehq/apps/app_manager/tests/util.py]: what's the reason behind this change?
these two now seem to be different things, my understanding is that `app.delete` would call other model callbacks but now this won't.

File: corehq/warehouse/etl.py [hunk 2]
```diff
@@ -125,3 +113,12 @@ def _render_template(path, context):
 
     template = engines['django'].from_string(template_string)
     return template.render(context)
+
+
```
Comment (mkangia) [inline corehq/warehouse/etl.py]: should this be `memoized`? i think it depends on the `slugs` that would not change unless there is a change in code.

File: corehq/warehouse/tests/test_validate.py [hunk 3]
```diff
@@ -3,6 +3,30 @@
...
-    for slug in ALL_TABLES:
+    def check(slug):
         loader = get_loader_by_slug(slug)
         loader().validate()
+
+    for slug in ALL_TABLES:
+        yield check, slug
+
+
+def check_for_loader_cycles(slug):
```
Comment (mkangia) [inline corehq/warehouse/tests/test_validate.py]: in what scenario will this test reveal an issue?

File: corehq/apps/app_manager/tests/util.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/app_manager/tests/util.py]: I'm not aware of any 'callbacks' we do on deletion of models. The reason for doing this is to avoid having to wrap the doc.

File: corehq/warehouse/tests/test_validate.py (hunk 3 above)
Comment (snopoke) [reply corehq/warehouse/tests/test_validate.py]: e.g. app_status_fact -> app_status_form_staging -> app_status_fact

It's not a real issue since these dependencies aren't really used for anything in code but it does help with understanding the flow without having to go and check airflow.

File: corehq/warehouse/etl.py (hunk 2 above)
Comment (snopoke) [reply corehq/warehouse/etl.py]: it could be but I don't think it's worth it since it's a cheap operation and it doesn't get called frequently

File: corehq/warehouse/management/commands/commit_table.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 class Command(BaseCommand):
     """
-    Example: ./manage.py stage_table group_staging 222617b9-8cf0-40a2-8462-7f872e1f1344
+    Example: ./manage.py stage_table group_staging 22
```
Comment (mkangia) [inline corehq/warehouse/management/commands/commit_table.py]: does that id need to be something like a UUID? i would be in favour of keeping the long example so that there is that hint of what it should be.

Comment (snopoke) [reply corehq/warehouse/management/commands/commit_table.py]: actually this used to be a UUID but is now an integer

Comment (mkangia) [reply corehq/warehouse/management/commands/commit_table.py]: Perfect.
